### ISSUE TITLE
Question: Run by line number

### ISSUE THREAD

**Issue by viktor-silakov** [vitalets/playwright-bdd#178]
Can I run a specific scenario by line number? It is a handy feature in cucumber-like tools. Example:

`npx wdio --spec ./features/test.feature:N`

where N - is the number of a scenario name line or line with a specific Example. Yes I know about the @only tag but in some situations run by line is more suitable

**Comment by alescinskis** [comment]
Why not give your scenarios custom tags (e.g. `@scenarioFailedLogin`) and then execute by a specific tag or groups of tags? I think it's more beneficial in the long run, than relying on line number that might change.

**Comment by viktor-silakov** [comment]
> Why not give your scenarios custom tags (e.g. `@scenarioFailedLogin`) and then execute by a specific tag or groups of tags? I think it's more beneficial in the long run, than relying on line number that might change.

You are right, this is not for the long run, for debugging mostly, it can be used for example to run certain scenarios via IDE extension (e.g. VScode) to run certain scenario. BTW there is no way to run certain example in Outline using the tag system.

**Comment by vitalets** [comment]
I saw this in Cucumber, and I also curious what is the dev flow for it?

Playwright itself does not provide such feature. It can run particular scenario [by file + scenario name](https://playwright.dev/docs/running-tests#run-specific-tests).
So, to get it working, we need Playwright to support it on the runner side. Or generate only that scenario in spec file.

For VSCode the expected way for debugging playwright-bdd tests is using Playwright VSCode extension, that allows to run particular scenario / example just by click:

![image](https://github.com/user-attachments/assets/9f6a6a2a-14f6-4259-bc79-8956fd8a4279)

**Comment by viktor-silakov** [comment]
> I saw this in Cucumber, and I also curious what is the dev flow for it?
<img width="201" alt="image" src="https://github.com/user-attachments/assets/94f2a5b9-bc5e-4401-b0a2-adf68d6bfec5">
<img width="255" alt="image" src="https://github.com/user-attachments/assets/4b149022-ca64-4ff3-9b90-18840abac6ae">

Very convenient to run certain example in such a way for example. Especially to run such things on ci/cd&.

**Comment by vitalets** [comment]
Ok.
Btw, do you the VSCode extension that allows to run example line from feature file just by click?

**Comment by viktor-silakov** [comment]
> Ok. Btw, do you the VSCode extension that allows to run example line from feature file just by click?

I use my own one for WDIO + Cucumber. But really it is more for CI/CD not for extension, there are cases when I need to do this on some heavy e2e outlines.
